util/ci/indent_tab_preprocess.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/python3
+
+import sys
+import re
+
+# preprocess files for indent check
+
+# Code processing:
+#
+# Tabs are kept.
+# Spaces after tabs are removed if indent is not smaller than in previous line.
+#
+# Example:
+#
+# <tab>a = 6;
+# <spc>b = 5;
+# <tab><spc>if (
+# <tab><tab><tab><spc><spc>) {
+# <tab><tab>c = 7;
+# <tab><spc>}
+#
+# Result:
+#
+# <tab>a = 6;
+# <spc>b = 5;
+# <tab>if (
+# <tab><tab><tab>) {
+# <tab><tab>c = 7;
+# <tab><spc>}
+
+# Multiline comments processing:
+#
+# It is expected that tabs indent from comment begining is
+# used as indent on begining of every commented line.
+#
+# Example:
+#
+# <tab>/*
+# <spc><spc>A
+# <tab><tab>B
+# <tab><spc>C
+# D
+# <tab><spc>*/
+# <tab><tab>/*
+# <spc><tab><spc>R
+#	<tab><spc><tab>*/
+#
+# Result:
+#
+# <tab>/*
+# <spc>A
+# <tab>B
+# <tab>C
+# D
+# <tab>*/
+# <tab><tab>/*
+# <spc><tab>*R
+#	<tab><spc>*/
+
+
+def main():
+	if len(sys.argv) != 5:
+		#print("Bad arguments")
+		#print(sys.argv)
+		exit()
+	# multiline comment begining and end from command line
+	mlc_begin = sys.argv[1]
+	mlc_end = sys.argv[2]
+	mlc_end_len = int(sys.argv[3])
+	file = open(sys.argv[4], "r")
+
+	re_mlc_begin = "^.*{}".format(mlc_begin)
+	re_mlc_begin_end = "^.*{}.*{}".format(mlc_begin, mlc_end)
+	re_mlc_end = "^.*{}".format(mlc_end)
+
+	lines = []
+
+	#print(re_mlc_begin)
+	#print(re_mlc_begin_end)
+	#print(re_mlc_end)
+	#print(file)
+
+	# -1 means not in the multiline comment
+	# other values mean multiline comment indent in the number of tabs
+	limit = -1
+	# this value stores thr previous line indent in code mode (not in multiline comment)
+	prev = 80
```

Comment:
It will cause the spaces after the tabs on the first line to be ignored.
So errors like in [src/unittest/test_random.cpp](https://github.com/minetest/minetest/pull/15002/files#diff-79e35cbfe7408d8e21e7f9bc27ef68e49b940299dd80c619e6a37619fcfc5dc3) will not be reported;

First line:
```
/* block comment on file beginning without space indent
 /* block comment on file beginning with space indent
```